tests/presets/cheatcode/Cheatcode.t.sol

```diff
@@ -0,0 +1,247 @@
+
+    // test expected emits
+    event Something(uint256 indexed topic1, uint256 indexed topic2, uint256 indexed topic3, uint256 data);
+    event SomethingNonIndexed(uint256 data);
+
+    function test() external {
+        string memory s = string(abi.encodePacked(block.timestamp));
+        address randomAddr = makeAddr(s);
+        uint256 random = uint160(randomAddr);
+        bytes32 randomBytes = bytes32(random);
+        bytes memory contractCode = bytes("0x6080");
+        VmSafe.CallerMode callerMode;
+        address msgSender;
+        address txOrigin;
+        address oldSender = msg.sender;
+        address oldOrigin = tx.origin;
+
+        assertNotEq(block.timestamp, random);
+        vm.warp(random);
+        assertEq(block.timestamp, random);
+
+        assertNotEq(block.number, random);
+        vm.roll(random);
+        assertEq(block.number, random);
+
+        assertNotEq(block.basefee, random);
+        vm.fee(random);
+        assertEq(block.basefee, random);
+
+        assertNotEq(block.prevrandao, random);
+        vm.prevrandao(randomBytes);
+        assertEq(block.prevrandao, random);
+
+        assertNotEq(block.chainid, 100);
+        vm.chainId(100);
+        assertEq(block.chainid, 100);
+
+        assertNotEq(tx.gasprice, random);
+        vm.txGasPrice(random);
+        assertEq(tx.gasprice, random);
+
+        assertNotEq(block.coinbase, randomAddr);
+        vm.coinbase(randomAddr);
+        assertEq(block.coinbase, randomAddr);
+
+        assertNotEq(randomBytes, vm.load(randomAddr, slot0));
+        vm.store(randomAddr, slot0, randomBytes);
+        assertEq(randomBytes, vm.load(randomAddr, slot0));
+
+        assertEq(0, randomAddr.balance);
+        vm.deal(randomAddr, 100);
+        assertEq(100, randomAddr.balance);
+
+        assertEq(0, address(randomAddr).code.length);
+        vm.etch(randomAddr, contractCode);
+        assertEq(contractCode, address(randomAddr).code);
+
+        // Test readCallers, prank
+
+        // readCallers before prank
+        (callerMode, msgSender, txOrigin) = vm.readCallers();
+        assertEq(uint256(callerMode), uint256(VmSafe.CallerMode.None));
+        assertEq(msgSender, oldSender);
+        assertEq(txOrigin, oldOrigin);
+        // Only prank msg.sender
+        vm.prank(randomAddr);
+        (callerMode, msgSender, txOrigin) = vm.readCallers();
+        assertEq(uint256(callerMode), uint256(VmSafe.CallerMode.Prank));
+        assertEq(msgSender, randomAddr);
+        assertEq(txOrigin, oldOrigin);
+        // Consume the prank
+        (bool _success,) = randomAddr.call(abi.encodeWithSignature("withdraw()"));
+        (callerMode, msgSender, txOrigin) = vm.readCallers();
+        assertEq(uint256(callerMode), uint256(VmSafe.CallerMode.None));
+        assertEq(msgSender, oldSender);
+        assertEq(txOrigin, oldOrigin);
+        // Prank msg.sender and tx.origin
+        vm.prank(randomAddr, randomAddr);
```

Comment:
Using another contract to make sure prank is working:

require(msg.sender==0x100)
require(tx.origin==0x200)